### ISSUE TITLE
Question: Interested in a new 'sender'?

### ISSUE THREAD

**Issue by sqall01** [mendhak/gpslogger#652]
Hi,

first of all I just forked your project and got it to run. I like it very much. Thank you for providing it.

I am currently working on an end-to-end encrypted GPS tracking service. This service aims to provide the user with the ability to store all GPS data on this service by still protecting their privacy (because it was encrypted on the device and the server does not know anything about the data and key). The code for the service will be open-sourced and anyone who wants to self-host it, is able to do this.

At the moment I have written the server, the web interface and a client based on python (which can be run on a Raspberry Pi for example). Now I wanted to write an Android App that tracks the GPS data and sends it to the server. And I thought why should I start from scratch when someone else did already an amazing job in writing the GPS tracking part? So I found your app and it fits perfectly. So, I was wondering if you are interested in a new sender in your app? I would write the code and would make a pull request afterwards. Is this fine by you and are you interested?

Cheers,
sqall

**Comment by mendhak** [comment]
Hi thanks for your work - I'd be interested in seeing what the implementation looks like but not merging it into the main codebase - since it's quite niche compared to what most of the users are doing.  For now anyway.

But what I'd be looking at is what you're encrypting and possibly incorporate that into the Custom URL logger. 

Another idea that came to mind when I read your post... if GPSLogger had a plugin model then as long as the interface is implemented then any kind of logger or sender could be written.  But for this I'd need to explore how separable the existing loggers and senders are, and if any hardcoded assumptions exist.

**Comment by sqall01** [comment]
I completely understand. I will still start writing the 'sender' for GPSlogger anyway. When I am finished with the whole project maybe you want to give it a try? Perhaps you like it enough and are ok with integrating the 'sender' into the codebase because you see benefits for some of your users.

However, if I have questions regarding the codebase while I am writing the 'sender', is it ok for me to ask you about it?

**Comment by mendhak** [comment]
Yes sure just post any questions here.  Though I may be slow to respond sometimes (and by sometimes I mean most of the time)

**Comment by christianTF** [comment]
@sqall01 Hi Andre,
I am just looking for a real time receiver with web interface, running cloud-based or locally on a Raspberry.
As I am GPS-logging from a motor bike, my intention is to first collect all routes, but primary can give wife and friends controlled access to my current track and location for the case of an accident, and to provide live data on tour (like you can share your drive in Waze).

Keep us informed about the status of your project!

Christian

**Comment by sqall01** [comment]
Hi @christianTF ,
thanks for your interest. I am pleased to hear this :) . This is actually exactly the usage I had in mind. For example, I want to place an old phone in my car and constantly track it live (since my wife and I are sharing the car, coordinating gets easier). Additionally, I had an idea of building a silent alarm for my car (sending me a notification when it leaves and is not allowed to).

I will definitely keep you informed :)

Andre

**Comment by sqall01** [comment]
Hi,
it took me a while but have finished the first version of an end-to-end encrypted GPS tracking system. I based the Android tracking part on a simpler logging application which had less features. The tracking system can either be self-hosted or used as a service (by registering a free account).

If you are still interested in this project (or perhaps even adding a sender to your application), just let me know.

Anyway, here are the relevant links for it:

[https://github.com/sqall01/chasr-android-logger](Android Logger)
[https://alertr.de/chasr/](Service)
[https://github.com/sqall01/chasr-server](Server)

Cheers,
Andre
